src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java

```diff
@@ -275,14 +269,21 @@ public Void run() throws InitializationException {
         }
     }
 
-    private AbstractReloadingMetadataResolver createMetadataResolver(final Settings settings, final Path configPath)
+    private AbstractBatchMetadataResolver createMetadataResolver(final Settings settings, final Path configPath)
             throws Exception {
-        final AbstractReloadingMetadataResolver metadataResolver;
+        final AbstractBatchMetadataResolver metadataResolver;
 
+        final String idpMetadataUrl = settings.get("idp.metadata_url");
+        final String idpMetadataFile = settings.get("idp.metadata_file");
+        final String idpMetadataBody = settings.get("idp.metadata_body");
         if (idpMetadataUrl != null) {
-            metadataResolver = new SamlHTTPMetadataResolver(settings, configPath);
+            metadataResolver = new SamlHTTPMetadataResolver(idpMetadataUrl, settings, configPath);
+        } else if (idpMetadataFile != null) {
+            metadataResolver = new SamlFilesystemMetadataResolver(idpMetadataFile, settings, configPath);
+        } else if (idpMetadataBody != null) {
+            metadataResolver = new SamlDOMMetadataResolver(idpMetadataBody);
```

Comment:
Use `DOMMetadataResolver`:
```suggestion
            metadataResolver = new DOMMetadataResolver(getMetadataDOM(idpMetadataBody));
```